Python/Tests/Utilities.Python.Analysis/PythonPaths.cs

```diff
@@ -215,21 +241,32 @@ public static class PythonVersionExtensions {
                     Assert.AreEqual(0, output.ExitCode);
                 }
             }
+            return envPath;
         }
 
         /// <summary>
         /// Creates a Python virtual environment in specified directory.
         /// </summary>
-        public static void CreateVirtualEnv(this PythonVersion pyVersion, string envPath) {
-            var virtualEnvModule = (pyVersion.Version < PythonLanguageVersion.V30) ? "virtualenv" : "venv";
-            using (var p = ProcessOutput.RunHiddenAndCapture(pyVersion.InterpreterPath, "-m", virtualEnvModule, envPath)) {
-                Console.WriteLine(p.Arguments);
-                Assert.IsTrue(p.Wait(TimeSpan.FromMinutes(3)));
-                Console.WriteLine(string.Join(Environment.NewLine, p.StandardOutputLines.Concat(p.StandardErrorLines)));
-                Assert.AreEqual(0, p.ExitCode);
+        public static string CreateVirtualEnv(this PythonVersion pyVersion, VirtualEnvName envName, string rootDirectory = null) {
```

Comment:
This change here is to speed up tests. Creating VirtualEnvs takes like 10 seconds each when tests can share the same virtual env.

Reply:
I thought I had a change similar to this in testing.

Reply:
this was the change. https://github.com/microsoft/PTVS/pull/5505

Reply:
I believe that change would cache the source and results folders. Virtual Envs are being created in the temp folder.

Reply:
Well at least some of them are.